clicache/integration-test/ThinClientPdxTests2.cs

```diff
@@ -904,8 +907,18 @@ private void accessPdxInstance()
       PdxType.GenericValCompare((long) ret.GetField("m_long"), dPdxType.Long);
       PdxType.GenericCompare((int[]) ret.GetField("m_int32Array"), dPdxType.Int32Array);
 
-      PdxType.GenericCompare((long[]) ret.GetField("m_ulongArray"), dPdxType.UlongArray);
-      PdxType.GenericCompare((short[]) ret.GetField("m_uint16Array"), dPdxType.Uint16Array);
+      string retUInt16ArrayStr = String.Join(",", ((short[])ret.GetField("m_uint16Array")).Select(p => ((ushort)(short)p).ToString()).ToArray());
+      string dPdxTypeUShortArrayStr = String.Join(",", dPdxType.Uint16Array.Select(p => p.ToString()).ToArray());
+      PdxType.GenericValCompare(retUInt16ArrayStr, dPdxTypeUShortArrayStr);
+
+      string retUintArrayStr = String.Join(",", ((int[])ret.GetField("m_uint32Array")).Select(p => ((uint)(int)p).ToString()).ToArray());
+      string dPdxTypeUintArrayStr = String.Join(",", dPdxType.Uint32Array.Select(p => p.ToString()).ToArray());
+      PdxType.GenericValCompare(retUintArrayStr, dPdxTypeUintArrayStr);
+
+      string retUlongArrayStr = String.Join(",", ((long[])ret.GetField("m_ulongArray")).Select(p => ((ulong)(long)p).ToString()).ToArray());
+      string dPdxTypeUlongArrayStr = String.Join(",", dPdxType.UlongArray.Select(p => p.ToString()).ToArray());
+      PdxType.GenericValCompare(retUlongArrayStr, dPdxTypeUlongArrayStr);
+
```

Comment:
Yah, the full story. Some consider ulong (same as ULONG), which are 64 bits in .NET, to add confusion. I actually like it. It's C++'s long long and unsigned long long that I dislike. Where Microsoft screwed up is making long and unsigned long only 32 bits in Windows: "... The true "war" was for sizeof(long), where Microsoft decided for sizeof(long) == 4 (LLP64) while nearly everyone else decided for sizeof(long) == 8 (LP64)."

There are no unsigned types in Java. Period. So there was no unsigned support for them in Pdx. This PR fixes that. Not just for the primitives, but for the arrays of the primitives.

Reply:
Regarding Java interop with unsigned types, I believe the only issue is if you try to use an unsigned type that exceeds the range of the signed type. For example, 16 bit integers go from -32768 to 32767 for signed, and 0 to 65535 for unsigned. So if you put a value of 50K from C# for example, this shows up as a -4350 in Java. So would be an access violation if you tried to use that as an index into an array in your Java code. As a minimum, we should document this in our Java interop docs (if we have such a thing). I believe the main problem we're trying to solve here is to provide support for the .NET unsigned types. People use unsigned types because they are natural (think size_t), not because they double the range.